[Android] Unable to destroy activity on startSingleScreenApp

### Issue Description
Hey ! 

I just encountered a bug on android when I try to call `startSingleScreenApp` from a modal. See the stacktrace:
```
Process: com.example, PID: 8472
java.lang.RuntimeException: Unable to destroy activity {com.example/com.reactnativenavigation.controllers.NavigationActivity}: java.util.EmptyStackException
   at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4366)
   at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4384)
   at android.app.ActivityThread.-wrap6(ActivityThread.java)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1588)
   at android.os.Handler.dispatchMessage(Handler.java:102)
   at android.os.Looper.loop(Looper.java:154)
   at android.app.ActivityThread.main(ActivityThread.java:6290)
   at java.lang.reflect.Method.invoke(Native Method)
   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
Caused by: java.util.EmptyStackException
   at java.util.Stack.peek(Stack.java:102)
   at com.reactnativenavigation.screens.ScreenStack.peek(ScreenStack.java:233)
   at com.reactnativenavigation.layouts.SingleScreenLayout.getCurrentScreen(SingleScreenLayout.java:295)
   at com.reactnativenavigation.controllers.Modal.dismiss(Modal.java:175)
   at com.reactnativenavigation.controllers.ModalController.destroy(ModalController.java:83)
   at com.reactnativenavigation.controllers.NavigationActivity.destroyLayouts(NavigationActivity.java:157)
   at com.reactnativenavigation.controllers.NavigationActivity.onDestroy(NavigationActivity.java:149)
   at android.app.Activity.performDestroy(Activity.java:6971)
   at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1169)
   at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4353)
   at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4384) 
   at android.app.ActivityThread.-wrap6(ActivityThread.java) 
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1588) 
   at android.os.Handler.dispatchMessage(Handler.java:102) 
   at android.os.Looper.loop(Looper.java:154) 
   at android.app.ActivityThread.main(ActivityThread.java:6290) 
   at java.lang.reflect.Method.invoke(Native Method) 
   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```
### Steps to Reproduce / Code Snippets / Screenshots
It is caused by the call of `getCurrentScreen` on Modal.js :
```java
NavigationApplication.instance.getEventEmitter().sendScreenChangedEvent("willDisappear", layout.getCurrentScreen().getNavigatorEventId());
NavigationApplication.instance.getEventEmitter().sendScreenChangedEvent("didDisappear", layout.getCurrentScreen().getNavigatorEventId());
```
In fact, it tries to get the current screen but the stack is empty so I tried an ugly fix by wrapping those calls in a try catch statement to see if it fixes the problem and it does. 

I was able to identify the commit causing this problem : https://github.com/wix/react-native-navigation/commit/763b26da6c5314d05526c610664141a843279eb4
I don't know the best way to handle this use case. @guyca could you give me a hand on this ?

Thanks !

---
### Environment
* React Native Navigation version: 1.1.30
* React Native version: 0.42.0
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Physical device, Marshmallow, Debug & Release

--- a/android/app/src/main/java/com/reactnativenavigation/controllers/Modal.java
+++ b/android/app/src/main/java/com/reactnativenavigation/controllers/Modal.java
@@ -172,8 +172,10 @@ public class Modal extends Dialog implements DialogInterface.OnDismissListener, 
 
     @Override
     public void dismiss() {
-        NavigationApplication.instance.getEventEmitter().sendScreenChangedEvent("willDisappear", layout.getCurrentScreen().getNavigatorEventId());
-        NavigationApplication.instance.getEventEmitter().sendScreenChangedEvent("didDisappear", layout.getCurrentScreen().getNavigatorEventId());
+        if (!isDestroyed) {
+            NavigationApplication.instance.getEventEmitter().sendScreenChangedEvent("willDisappear", layout.getCurrentScreen().getNavigatorEventId());
+            NavigationApplication.instance.getEventEmitter().sendScreenChangedEvent("didDisappear", layout.getCurrentScreen().getNavigatorEventId());
+        }
         super.dismiss();
     }
 

